### PR TITLE
Added README.md for demo_nodes_py

### PR DESCRIPTION
**Pull request by cardboardcode**
## Purpose of Pull Request

This pull request adds a `README.md` to **demo_nodes_py** as part of efforts to address the issue highlighted in https://github.com/ros2/demos/issues/531.

### CODE REVIEW

File: demo_nodes_py/README.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+[INFO] [1674574042.992465274] [async_param_client]:         successful: True
+[INFO] [1674574042.992703259] [async_param_client]:         reason:
+[INFO] [1674574042.992941566] [async_param_client]:     string_parameter:
+[INFO] [1674574042.993179508] [async_param_client]:         successful: True
+[INFO] [1674574042.993416599] [async_param_client]:         reason:
+```
+
+### Serialized Subscriber
+
+Running `ros2 run demo_nodes_py talker` should have string printed to termina in a similar manner:
```
Comment (mjeronimo) [inline demo_nodes_py/README.md:270]:
```suggestion
Running `ros2 run demo_nodes_py talker` should print output to the terminal similar to the following: 
```

File: demo_nodes_py/README.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+```bash
+# Open new terminal
+ros2 run demo_nodes_py listener
+```
+
+![](img/talker_listener.png)
+
+### Server & Client
+
+This runs `add_two_ints_client` and `add_two_ints_server` ROS 2 client and server which Server processes two integers sent from Client and publishes its sum to Client to be printed out.
```
[the remark this answers is not in the document]
Comment (cardboardcode) [reply demo_nodes_py/README.md:48]: Done.

File: demo_nodes_py/README.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+Double value is: 28.0
+```
+Run `ros2 param get /set_parameters_callback param2` should print the following to terminal:
+```bash
+Double value is 4.0
+```
+
+### Asynchronous Parameter Client
+
+Running `ros2 run demo_nodes_cpp parameter_blackboard` should have similar strings printed to terminal similar to what is shown below:
```
[the remark this answers is not in the document]
Comment (cardboardcode) [reply demo_nodes_py/README.md:221]: Done.

File: demo_nodes_py/README.md (hunk 1 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:270]: Done.

File: demo_nodes_py/README.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
+## **What Is This?**
+
+This demo ...
+
+This package consists of ROS 2 nodes:
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:5]: Can remove "This demo ...",  and change the next line to,
"This package contains the following ROS 2 nodes:"

File: demo_nodes_py/README.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+```bash
+colcon build --packages-select demo_nodes_py
+```
+
+## **Run**
+
+### Basic Listener & Talker
+
+This runs `talker` and `listener` ROS 2 nodes which exchanges the following string with incremeting integer:
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:30]: "...which exchange the following..." , remove the extra 's'

File: demo_nodes_py/README.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+```bash
+# Open new terminal
+ros2 run demo_nodes_py listener
+```
+
+![](img/talker_listener.png)
+
+### Server & Client
+
+This runs `add_two_ints_client` and `add_two_ints_server` ROS 2 client and server where  Server processes two integers sent from Client and publishes its sum to Client to be printed out.
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:48]: Can remove the capitalization of "Server" and "Client". Also, "... where the server processes..." (add 'the')

File: demo_nodes_py/README.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py add_two_ints_client_async
+```
+![](img/server_client.png)
+
+### QoS Listener & Talker
+
+Similar to previous, this runs `talker_qos` and `listener_qos` ROS 2 nodes which exchanges using the Quality of Service (QoS) Best Effort policy.
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:74]: "... which exchange messages using the Quality of Service (QoS) best effort policy."

File: demo_nodes_py/README.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+# In terminal running listener
+[INFO] [1674573025.634810166] [listener]: I heard: [Hello World: 0]
+[INFO] [1674573026.596392653] [listener]: I heard: [Hello World: 1]
+[INFO] [1674573027.596400384] [listener]: I heard: [Hello World: 2]
+#...
+```
+
+### Basic Server & Client
+
+When executed correctly, strings should be printed to terminal similar to what is shown below:
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:150]: "...printed to the terminal ..."

File: demo_nodes_py/README.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+
+#### Client [Asynchronous]
+```bash
+# In terminal running add_two_ints_client_async
+[INFO] [1674553718.598690033] [add_two_ints_client]: Result of add_two_ints: 5
+```
+
+### QoS Listener & Talker
+
+When executed correctly, strings should be printed to terminal similar to what is shown below:
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:173]: "...printed to the terminal ...."

File: demo_nodes_py/README.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+Double value is: 28.0
+```
+Run `ros2 param get /set_parameters_callback param2` should print the following to terminal:
+```bash
+Double value is 4.0
+```
+
+### Asynchronous Parameter Client
+
+Running `ros2 run demo_nodes_cpp parameter_blackboard` should print output to the terminal similar to what is shown below:
```
Comment (adityapande-1995) [inline demo_nodes_py/README.md:220]: "... print an output ....."

**Review by adityapande-1995 — CHANGES_REQUESTED** [review]
Thanks for adding the README ! Looks good after the following nits resolved.

File: demo_nodes_py/README.md (hunk 4 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:5]: Thank you for the review. Will follow up with the nits resolved shortly.

Comment (cardboardcode) [reply demo_nodes_py/README.md:5]: Done.

File: demo_nodes_py/README.md (hunk 5 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:30]: Done.

File: demo_nodes_py/README.md (hunk 6 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:48]: Done. Added 'the' in front of 'client' as well.

Changed to "... publishes its sum to the client" and "sent from the client".

File: demo_nodes_py/README.md (hunk 7 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:74]: Done.

File: demo_nodes_py/README.md (hunk 8 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:150]: Done.

File: demo_nodes_py/README.md (hunk 9 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:173]: Done.

File: demo_nodes_py/README.md (hunk 10 above)
Comment (cardboardcode) [reply demo_nodes_py/README.md:220]: Done.

**Review by adityapande-1995 — APPROVED** [review]

**Pull request merged**